Assets/_BuildingBlocks/Credentials/Scripts/CredentialHandler.cs

```diff
@@ -0,0 +1,155 @@
+            ICredentialsPropertySection view = GetComponentInChildren<ICredentialsPropertySection>();
+            view.Handler = this;
+
+            keyVault.OnAuthorizationTypeDetermined.AddListener(DeterminedAuthorizationType);
+        }
+
+        private void OnDestroy()
+        {
+            keyVault.OnAuthorizationTypeDetermined.RemoveListener(DeterminedAuthorizationType);
+        }
+
+        public void UpdateUrl(string url)
+        {
+            if (!string.IsNullOrEmpty(url))
+            {
+                this.urlWithoutCredentials = url;
+                UrlHasChanged(url);
+            }
+        }
+
+        private void UrlHasChanged(string newURL)
+        {
+            this.urlWithoutCredentials = newURL;
+            keyVault.TryToFindSpecificCredentialType(newURL.ToString(), "");
+        }
+
+        private void DeterminedAuthorizationType(string url, AuthorizationType authorizationType)
+        {
+            this.authorizationType = authorizationType;
+            //we check if the authorized type is different from unkown. The keyvault webrequests can never return unknown if authorization was valid
+            hasAuthorizationType = authorizationType != AuthorizationType.Unknown;
+
+            //Debug.Log("Determined authorization type: " + authorizationType + " for url: " + url, this.gameObject);
+            if (hasAuthorizationType)
+            {
+                if(ConstructURLWithKey())
+                    CredentialsSucceeded?.Invoke(credentialUrl);
+            }
+        }
+
+        public void ApplyCredentials()
+        {
+            switch (authorizationType)
+            {
+                case AuthorizationType.UsernamePassword:
+                    keyVault.TryBasicAuthentication(
+                        urlWithoutCredentials,
+                        UserName,
+                        PasswordOrKeyOrTokenOrCode
+                    );
+                    break;
+                case AuthorizationType.InferableSingleKey:
+                    keyVault.TryToFindSpecificCredentialType(
+                        urlWithoutCredentials,
+                        PasswordOrKeyOrTokenOrCode
+                    );
+                    break;               
+            }
+        }
+
+        public void SetAuthorizationInputType(AuthorizationType type)
+        {
+            if (
+                type == AuthorizationType.Key
+                || type == AuthorizationType.Token
+                || type == AuthorizationType.BearerToken
+                || type == AuthorizationType.Code
+            )
+                type = AuthorizationType.InferableSingleKey;
+
+            authorizationType = type;
+        }
+
+        public void ClearKeyFromURL()
+        {
+            if (CredentialQuery != string.Empty)
+            {
+                credentialUrl = credentialUrl.Replace(CredentialQuery, string.Empty);
```

Comment:
Dit is typisch iets wat je ook mooi in een QueryStringCredential value object kan doen